modyn/supervisor/internal/pipeline_executor/pipeline_executor.py

```diff
@@ -277,13 +279,33 @@ def _handle_triggers_within_batch(self, batch: list[tuple[int, int, int]], trigg
 
             num_samples_in_trigger = self.grpc.get_number_of_samples(self.pipeline_id, trigger_id)
             if num_samples_in_trigger > 0:
+                if i > 0:
+                    # since num_samples_in_trigger is not 0, we are sure that triggering_data is not empty
+                    first_timestamp = triggering_data[0][1]
+                    last_timestamp = triggering_data[-1][1]
+                else:
+                    # now it is the first trigger in this batch. Triggering_data can be empty.
```

Comment:
How can `triggering_data` be empty if `num_samples_in_trigger > 0`? That confuses me a bit. And why can this only happen in the first trigger?

Reply:
so basically i am a bit confused why it's not always the code path for i > 0 and why this here can happen

Reply:
I answer everything in the unit test Q & A